bin/functions

```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env bash
-set -Eeuo pipefail
+set -e
+set -Euo pipefail
```

Comment:
I've separated `set -e` and the other options so that the script also exits when setting these options failed.

Reply:
Are you guarding against `set -Euo pipefail` failing? If so, what could cause that to fail?

Reply:
@halcyon Good question! It's guarding against running the script with a shell (not necessarily bash) which doesn't support `pipefail`.

With everything in one line (`set -Eeuo pipefail`), this command would fail and emit an error message but the script wouldn't be terminated.
With `set -e` executed before the other settings, the script would fail on `set -Euo pipefail` and be properly terminated.

Example:
https://askubuntu.com/questions/886537/set-e-o-pipefail-not-working-on-bash-script-on-ubuntu-16

Reply:
@joschi How could this script be run as anything but bash? `asdf` itself is invoked as a bash script due to its shebang. Even our `functions` script has a bash shebang. The example from askubuntu has a non-bash shebang.